<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>tda.Rmd</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cerulean.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
}

</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html"></a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Project Overview</a>
</li>
<li>
  <a href="data_exploration.html">Data Exploration</a>
</li>
<li>
  <a href="tda.html">Topological Data Analysis</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">tda.Rmd</h1>

</div>


<pre class="r"><code>library(ecodist, warn.conflicts = FALSE)
library(ggplot2, warn.conflicts = FALSE)
library(MASS, warn.conflicts = FALSE)
library(TDA, warn.conflicts = FALSE)
library(TDAstats, warn.conflicts = FALSE)
library(phyloseq, warn.conflicts = FALSE)
library(phylosmith, warn.conflicts = FALSE)
library(RColorBrewer, warn.conflicts = FALSE)
library(tidyverse, warn.conflicts = FALSE)
library(TDA, warn.conflicts = FALSE)
library(&quot;TDAstats&quot;, warn.conflicts = FALSE)
library(vegan, warn.conflicts = FALSE)</code></pre>
<div id="persistent-homology" class="section level1">
<h1>Persistent homology</h1>
<p>We’ll begin our exploration of topological data analysis with the application of persistent homology. We’ll start on the soil column data set, as there has been a clear signal demonstrated in NMDS plots indicating separation between soil-treated and manure-treated columns. Persistent homology should be able to detect some difference in the topological structure of two different treatments. After this verification, we will move on to applying persistent homology to our data set.</p>
<div id="soil-column" class="section level2">
<h2>Soil column</h2>
<p>We’ll begin by reading in the soil column data sets.</p>
<pre class="r"><code>soil_column.args &lt;- readRDS(&quot;./data/raw/arg-phyloseq.RDS&quot;)
soil_column.args</code></pre>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 355 taxa and 56 samples ]
## sample_data() Sample Data:       [ 56 samples by 7 sample variables ]
## tax_table()   Taxonomy Table:    [ 355 taxa by 2 taxonomic ranks ]</code></pre>
<pre class="r"><code>soil_column.args.cooccur &lt;- FastCoOccur(soil_column.args, &quot;Treatment&quot;)</code></pre>
<pre class="r"><code>phyloseq_ggplot_NMDS &lt;- function(phyloseq_obj, treatment){
  require(ggplot2); require(RColorBrewer); require(vegan); require(MASS); require(phylosmith)
  getPalette = colorRampPalette(brewer.pal(8, &quot;Dark2&quot;)); colorCount = 1 + length(unlist(unique(sample_data(phyloseq_obj)[[treatment]]))); colors = getPalette(colorCount); theme_set(theme_bw())

  MDS &lt;- metaMDS(t(otu_table(find_generalists(phyloseq_obj))), autotransform = FALSE, distance = &quot;bray&quot;, k=3, trymax=50)
  plot(MDS, display = c(&quot;sites&quot;, &quot;species&quot;), choices = c(1,2), type = &quot;p&quot;)
  abline(h=0,lty=2)
  abline(v=0,lty=2)
  stressplot(MDS)

  Treatment &lt;- sample_data(phyloseq_obj)[[treatment]]
  MDS$points[,&#39;MDS1&#39;]
  MDS1 &lt;- data.frame(scores(MDS))$NMDS1
  MDS2 &lt;- data.frame(scores(MDS))$NMDS2
  NMDS &lt;- data.frame(MDS1,MDS2,Treatment)
  NMDS.narm &lt;- subset(NMDS, !is.na(Treatment))
  veganCovEllipse&lt;-function (cov, center = c(0, 0), scale = 1, npoints = 100){
    theta &lt;- (0:npoints) * 2 * pi/npoints
    Circle &lt;- cbind(cos(theta), sin(theta))
    t(center + scale * t(Circle %*% chol(cov)))}
  df_ell &lt;- data.frame()
  for(g in unique(NMDS.narm[[treatment]])){
    df_ell &lt;- rbind(df_ell, cbind(as.data.frame(with(NMDS.narm[NMDS.narm[,treatment]==g,], veganCovEllipse(cov.wt(cbind(MDS1,MDS2),wt=rep(1/length(MDS1),length(MDS1)))$cov,center=c(mean(MDS1),mean(MDS2))))),group=g))}

  p &lt;- ggplot(data = NMDS.narm, aes(MDS1, MDS2)) +
    geom_point(aes(color = treatment), size=1.5, alpha=0.75) +
    geom_path(data=df_ell, aes(x=MDS1, y=MDS2, colour=group), size=2, linetype=5) +
    scale_color_manual(values=colors) +
    theme_classic() +
    theme(aspect.ratio=1,
          axis.line.x = element_line(colour = &#39;black&#39;, size = 1, linetype = &#39;solid&#39;),
          axis.line.y = element_line(colour = &#39;black&#39;, size = 1, linetype = &#39;solid&#39;),
          axis.text.x = element_text(size = 10, face = &quot;bold&quot;),
          axis.text.y = element_text(size = 10, face = &quot;bold&quot;),
          axis.title.x = element_text(size = 12, face = &quot;bold&quot;),
          axis.title.y = element_text(size = 12, face = &quot;bold&quot;),
          legend.title = element_blank(),
          legend.text = element_text(size = 11, face = &quot;bold&quot;),
          legend.background = element_rect(fill = (alpha = 0))
    )
  return(p)
}
phyloseq_ggplot_NMDS(soil_column.args, &quot;Treatment&quot;)</code></pre>
<pre><code>## Run 0 stress 0.05503205 
## Run 1 stress 0.05650123 
## Run 2 stress 0.05502962 
## ... New best solution
## ... Procrustes: rmse 0.005412249  max resid 0.01895538 
## Run 3 stress 0.05507145 
## ... Procrustes: rmse 0.00316764  max resid 0.01660079 
## Run 4 stress 0.05509609 
## ... Procrustes: rmse 0.003658389  max resid 0.01645545 
## Run 5 stress 0.05505899 
## ... Procrustes: rmse 0.00635836  max resid 0.02007478 
## Run 6 stress 0.05502996 
## ... Procrustes: rmse 0.004941623  max resid 0.01430185 
## Run 7 stress 0.05510626 
## ... Procrustes: rmse 0.006112046  max resid 0.0211802 
## Run 8 stress 0.05508341 
## ... Procrustes: rmse 0.003632519  max resid 0.02308273 
## Run 9 stress 0.05506989 
## ... Procrustes: rmse 0.007225719  max resid 0.02237372 
## Run 10 stress 0.05646389 
## Run 11 stress 0.05647346 
## Run 12 stress 0.05504061 
## ... Procrustes: rmse 0.006094545  max resid 0.020649 
## Run 13 stress 0.0550578 
## ... Procrustes: rmse 0.003226354  max resid 0.01317222 
## Run 14 stress 0.05503616 
## ... Procrustes: rmse 0.005706899  max resid 0.0195141 
## Run 15 stress 0.05502547 
## ... New best solution
## ... Procrustes: rmse 0.004760918  max resid 0.01790104 
## Run 16 stress 0.05627753 
## Run 17 stress 0.05646444 
## Run 18 stress 0.05510156 
## ... Procrustes: rmse 0.009976234  max resid 0.03080669 
## Run 19 stress 0.05627471 
## Run 20 stress 0.05628288 
## Run 21 stress 0.05647282 
## Run 22 stress 0.05503779 
## ... Procrustes: rmse 0.005934966  max resid 0.02191093 
## Run 23 stress 0.05627357 
## Run 24 stress 0.05510909 
## ... Procrustes: rmse 0.01053016  max resid 0.03327724 
## Run 25 stress 0.05629419 
## Run 26 stress 0.05511257 
## ... Procrustes: rmse 0.0106815  max resid 0.03362673 
## Run 27 stress 0.05503005 
## ... Procrustes: rmse 0.005110026  max resid 0.01839773 
## Run 28 stress 0.05506659 
## ... Procrustes: rmse 0.003414157  max resid 0.01982158 
## Run 29 stress 0.05511535 
## ... Procrustes: rmse 0.01091533  max resid 0.03439203 
## Run 30 stress 0.05516267 
## ... Procrustes: rmse 0.01315903  max resid 0.04066907 
## Run 31 stress 0.0550848 
## ... Procrustes: rmse 0.008582233  max resid 0.02714394 
## Run 32 stress 0.05646943 
## Run 33 stress 0.05508948 
## ... Procrustes: rmse 0.009029669  max resid 0.02889592 
## Run 34 stress 0.05629771 
## Run 35 stress 0.0550533 
## ... Procrustes: rmse 0.003429807  max resid 0.01946814 
## Run 36 stress 0.05502604 
## ... Procrustes: rmse 0.00167166  max resid 0.006945746 
## ... Similar to previous best
## *** Solution reached</code></pre>
<p><img src="tda_files/figure-html/unnamed-chunk-1-1.png" width="672" /><img src="tda_files/figure-html/unnamed-chunk-1-2.png" width="672" /><img src="tda_files/figure-html/unnamed-chunk-1-3.png" width="672" /></p>
<p>Performing NMDS, reducing data. Preparing data for persistent homology.</p>
<pre class="r"><code>soil_column.nmds &lt;- metaMDS(t(otu_table(find_generalists(soil_column.args))), autotransform = FALSE,
                            distance = &quot;bray&quot;, k = 2, trymax = 50)</code></pre>
<pre><code>## Run 0 stress 0.0724996 
## Run 1 stress 0.0725037 
## ... Procrustes: rmse 0.0009457564  max resid 0.005094322 
## ... Similar to previous best
## Run 2 stress 0.1090752 
## Run 3 stress 0.123617 
## Run 4 stress 0.07250204 
## ... Procrustes: rmse 0.000910602  max resid 0.005224942 
## ... Similar to previous best
## Run 5 stress 0.07250352 
## ... Procrustes: rmse 0.001030046  max resid 0.005328111 
## ... Similar to previous best
## Run 6 stress 0.0725046 
## ... Procrustes: rmse 0.0009577302  max resid 0.005082376 
## ... Similar to previous best
## Run 7 stress 0.07250217 
## ... Procrustes: rmse 0.0009040105  max resid 0.005185875 
## ... Similar to previous best
## Run 8 stress 0.0725032 
## ... Procrustes: rmse 0.0009248037  max resid 0.005121886 
## ... Similar to previous best
## Run 9 stress 0.07249989 
## ... Procrustes: rmse 0.0002746778  max resid 0.001030654 
## ... Similar to previous best
## Run 10 stress 0.07250214 
## ... Procrustes: rmse 0.0009039538  max resid 0.00518503 
## ... Similar to previous best
## Run 11 stress 0.1076221 
## Run 12 stress 0.123707 
## Run 13 stress 0.07250408 
## ... Procrustes: rmse 0.001064049  max resid 0.005366366 
## ... Similar to previous best
## Run 14 stress 0.1266631 
## Run 15 stress 0.1080577 
## Run 16 stress 0.1096588 
## Run 17 stress 0.1259284 
## Run 18 stress 0.1096451 
## Run 19 stress 0.1229827 
## Run 20 stress 0.1230967 
## *** Solution reached</code></pre>
<pre class="r"><code># soil_column.nmds &lt;- metaMDS(t(otu_table(find_generalists(soil_column.args))), autotransform = FALSE,
#                            distance = &quot;bray&quot;, k = 2, trymax = 50)
PCA_1 &lt;- data.frame(scores(soil_column.nmds))$NMDS1
PCA_2 &lt;- data.frame(scores(soil_column.nmds))$NMDS2
treatments &lt;- sample_data(soil_column.args)[[&quot;Treatment&quot;]]
soil_column.nmds.transform &lt;- data.frame(PCA_1, PCA_2, treatments)
soil_column.nmds.transform.narm &lt;- subset(soil_column.nmds.transform, !is.na(treatments))</code></pre>
<p>Separating data by category.</p>
<pre class="r"><code>soil_column.nmds.manure &lt;- soil_column.nmds.transform.narm %&gt;% 
  filter(treatments == &quot;Manure&quot;) %&gt;% 
  select(PCA_1, PCA_2) %&gt;% 
  as.matrix()
soil_column.nmds.control &lt;- soil_column.nmds.transform.narm %&gt;% 
  filter(treatments == &quot;Control&quot;) %&gt;% 
  select(PCA_1, PCA_2) %&gt;% 
  as.matrix()</code></pre>
<p>Persistent homology by treatment. First, the manure treatment:</p>
<pre class="r"><code>manure.hom &lt;- calculate_homology(soil_column.nmds.manure, dim = 2)
plot_persist(manure.hom)</code></pre>
<p><img src="tda_files/figure-html/tda.3-1.png" width="672" /></p>
<pre class="r"><code>plot_barcode(manure.hom)</code></pre>
<p><img src="tda_files/figure-html/unnamed-chunk-2-1.png" width="672" /></p>
<pre class="r"><code>plot(manure.hom)</code></pre>
<p><img src="tda_files/figure-html/unnamed-chunk-3-1.png" width="672" /></p>
<p>Persistent homology for control data:</p>
<pre class="r"><code>control.hom &lt;- calculate_homology(soil_column.nmds.control, dim = 2)
plot_persist(control.hom)</code></pre>
<p><img src="tda_files/figure-html/unnamed-chunk-4-1.png" width="672" /></p>
<pre class="r"><code>plot_barcode(control.hom)</code></pre>
<p><img src="tda_files/figure-html/unnamed-chunk-5-1.png" width="672" /></p>
<p>We’ll now normalize the barcode graphs so that we can make more direct comparison of homologies between treatment and contorl groups.</p>
<pre class="r"><code>plot_barcode(control.hom) +
  ggtitle(&quot;Persistent homology: control&quot;) +
  xlim(c(0, 4))</code></pre>
<pre><code>## Scale for &#39;x&#39; is already present. Adding another scale for &#39;x&#39;, which
## will replace the existing scale.</code></pre>
<p><img src="tda_files/figure-html/unnamed-chunk-6-1.png" width="672" /></p>
<pre class="r"><code>plot_barcode(manure.hom) +
  ggtitle(&quot;Persistent homology: manure&quot;) +
  xlim(c(0, 4))</code></pre>
<pre><code>## Scale for &#39;x&#39; is already present. Adding another scale for &#39;x&#39;, which
## will replace the existing scale.</code></pre>
<p><img src="tda_files/figure-html/unnamed-chunk-7-1.png" width="672" /></p>
<p>We now run a permutation test on the two datasets to confirm that the persistent homologies of the two datasets are distinct. For more information on significance testing on persistent homologies, see “Hypothesis testing for topological data analysis”</p>
<pre class="r"><code>control_manure.perm_test &lt;- permutation_test(soil_column.nmds.control, 
                                             soil_column.nmds.manure, 
                                             iterations = 10000)</code></pre>
<p>The p-value for 0-cycles:</p>
<pre class="r"><code>print(control_manure.perm_test[[1]]$pvalue)</code></pre>
<pre><code>## [1] 0.0017</code></pre>
<p>So we see that the the p-value is below the significance threshhold, indicating that the structure of the 0-cycles in the two data sets are distinct. On the other hand, for the 1-cycles:</p>
<pre class="r"><code>print(control_manure.perm_test[[2]]$pvalue)</code></pre>
<pre><code>## [1] 0.0695</code></pre>
<p>This is above the traditional signifiance threshold of 0.05 for signifiance testing. Without further coroborrating evidence, we can’t assume there is statistical difference between the two data sets.</p>
<p>We can also plot the null distribution for the n-cycles:</p>
<pre class="r"><code>ggplot(data.frame(val = control_manure.perm_test[[1]]$permvals), aes(val)) + geom_histogram(binwidth = 0.25) +
  labs(x = &quot;Wasserstein Distance&quot;, y = &quot;Count&quot;, 
       title = &quot;Null distribution for 0-cycles&quot;,
       subtitle = &quot;(n = 1000)&quot;) +
  theme(plot.title = element_text(hjust = 0.5), 
        plot.subtitle = element_text(hjust = 0.5)) +
  geom_vline(xintercept = control_manure.perm_test[[1]]$wasserstein, 
             colour = &quot;#BB0000&quot;, linetype = &quot;dashed&quot;) </code></pre>
<p><img src="tda_files/figure-html/unnamed-chunk-11-1.png" width="672" /></p>
<pre class="r"><code>ggplot(data.frame(val = control_manure.perm_test[[2]]$permvals), aes(val)) + geom_histogram(binwidth = 0.01) +
  labs(x = &quot;Wasserstein Distance&quot;, y = &quot;Count&quot;, title = &quot;Null distribution for 2-cycles&quot;,
       subtitle = &quot;(n = 1000)&quot;) +
  theme(plot.title = element_text(hjust = 0.5), plot.subtitle = element_text(hjust = 0.5)) +
  geom_vline(xintercept = control_manure.perm_test[[2]]$wasserstein, colour = &quot;#BB0000&quot;, linetype = &quot;dashed&quot;) </code></pre>
<p><img src="tda_files/figure-html/unnamed-chunk-12-1.png" width="672" /></p>
<p>I forgot what this is supposed to be :)</p>
<pre class="r"><code>ggplot(data = soil_column.nmds.transform.narm, aes(x = PCA_1, y = PCA_2, color = treatments)) +
  geom_point()</code></pre>
<p><img src="tda_files/figure-html/unnamed-chunk-13-1.png" width="672" /></p>
</div>
</div>

<p><br>
<br>

<strong><a href="https://github.com/pommevilla">Paul Villanueva</a></strong>
<br>
Ph.D. Student, Bioinformatics and Computational Biology<br>
Iowa State University.  Ames, IA.</p>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
